content/influxdb/v2.0/reference/flux/stdlib/monitor/notify.md

```diff
@@ -29,26 +29,51 @@ monitor.notify(
 ## Parameters
 
 ### endpoint
-A function that constructs and sends the notification to an endpoint.
+({{< req >}}) A function that constructs and sends the notification to an endpoint.
 
 _**Data type:** Function_
 
 ### data
-Data to append to the output.
-**InfluxDB populates notification data.**
+({{< req >}}) Notification data to append to the output.
+This data specifies which notification rule and notification endpoint to associate
+with the sent notification.
 
 _**Data type:** Record_
 
+The data record must contain the following fields:
+
+- _notification_rule_id
+- _notification_rule_name
+- _notification_endpoint_id
+- _notification_endpoint_name
+
 ## Examples
 
 ### Send a notification to Slack
 ```js
 import "influxdata/influxdb/monitor"
+import "influxdata/influxdb/secrets"
 import "slack"
 
-endpoint = slack.endpoint(name: "slack", channel: "#flux")
+token = secrets.get(key: "SLACK_TOKEN")
+
+endpoint = slack.endpoint(token: token)(mapFn: (r) => ({
+    channel: "Alerts",
+    text: r._message,
+    color: "danger"
+  }))()
+
+notification_data = {
+	_notification_rule_id: "0000000000000001",
+	_notification_rule_name: "example-rule-name",
+	_notification_endpoint_id: "0000000000000002",
+	_notification_endpoint_name: "example-endpoint-name",
+}
```

Comment:
When you're writing your own custom notification task, can you just pick arbitrary values for these? Is it a problem if you pick the same value some other task is already using? Can you use the same endpoint ID in multiple tasks if you're using the same package, URL, token, etc when calling `<package>.endpoint()`?

Reply:
I've added the following to this section:

> The InfluxDB monitoring and alerting system uses `monitor.notify()` to store information about sent notifications and automatically assigns these values. If writing a custom notification [task](/influxdb/v2.0/process-data/), use **arbitrary, but unique** values.

You can use existing IDs and names, but it will intermingle logs and other metrics with that probably shouldn't be intermingled.

Reply:
That's great, very helpful! Thanks!

Reply:
We recommend using...for logging purposes?